sim/examples/random-player-ai.ts

```diff
@@ -0,0 +1,157 @@
+/**
+ * Example random player AI.
+ *
+ * Pokemon Showdown - http://pokemonshowdown.com/
+ *
+ * @license MIT
+ */
+
+import {ObjectReadWriteStream} from '../../lib/streams';
+import {BattlePlayer} from '../battle-stream';
+import {PRNG, PRNGSeed} from '../prng';
+
+export class RandomPlayerAI extends BattlePlayer {
+	readonly move: number;
+	readonly mega: number;
+	readonly prng: PRNG;
+
+	constructor(
+		playerStream: ObjectReadWriteStream<string>,
+		options: {move?: number, mega?: number, seed?: PRNG | PRNGSeed | null } = {},
+		debug: boolean = false
+	) {
+		super(playerStream, debug);
+		this.move = options.move || 1.0;
+		this.mega = options.mega || 0;
+		this.prng = options.seed && !Array.isArray(options.seed) ? options.seed : new PRNG(options.seed);
+	}
+
+	receiveError(error: Error) {
+		if (!error.message.startsWith('[Invalid choice]')) throw error;
+		// Somehow we had a choice error - just let the simulator autochoose for us.
```

Comment:
I think this piece of documentation would be better if it specified that this case might happen if a Pokémon is afflicted by Imprison, Arena Trap, Magnet Pull or Shadow Tag.

Reply:
I think it also maybe happens with Struggle? It also conveniently papers over bugs (as we discovered in #5286...). But yeah, explaining why its sometimes needed is a good call.

Reply:
Not really. If the Pokémon must Struggle, the request object lists only ``Struggle`` in the possible moves.

Reply:
Ack.